bindings/CXX11/adios2/cxx11/KokkosView.h

```diff
@@ -30,24 +31,25 @@ struct memspace_kokkos_to_adios2<Kokkos::CudaSpace>
 } // namespace detail
 
 template <class T>
-class AdiosView
+class KokkosView : public AdiosView<T>
```

Comment:
The example I am pointing out in [https://github.com/anagainaru/ADIOS2-addons/tree/main/GPUAware/kokkos/code](https://github.com/anagainaru/ADIOS2-addons/tree/main/GPUAware/kokkos/code) uses directly `Kokkos::View` with `Put`/`Get` functions. 

I do not want the user to need to pass a `adios2::KokkosView(Kokkos::View)`

See godbolt here with your code: [https://godbolt.org/z/Efo75xEW8](https://godbolt.org/z/Efo75xEW8)

The level of indirection you are removing was allowing me to pass directly `Kokkos::View` that I was later recasting it to `AdiosView` because this is the objects only role. I can get away with not using the level of indirection (in which case there is no ICE) and allow `Put` and `Get` functions to receive an `AdiosView` (since it has a constructor that take only the `Kokkos::View` passing directly the `Kokkos::View` is fine) but then the user will have to specifically tell the data type:

```
        bpWriter.Put<float>(data, gpuSimData);
```

I want this code to work with the user passing:
```
        bpWriter.Put(data, gpuSimData);
```

I can share the godbolt codes with my solutions, but you can also use the example in the link above.